### ISSUE TITLE
How can I create a new Role?

### ISSUE THREAD

**Issue by FabioLiv** [gondor/openstack4j#624]
Hi all,
I'm developing a project by exploiting openstack4j APIs (v. 2.0.9). I need to create a new role so I tried (after authentication) to use a Builder like Users creation:

`public void role_create(String name) {` 
`org.openstack4j.model.identity.Role` `role` `=`   `os.identity().roles().create(Builders.role().name(name).build());`
`}`

When I noticed the error `The method create(String) in the type RoleService is not applicable for the arguments (Role)`, I supposed I just had to pass the 'name' parameter to the create method:

`public` `void` `role_create(String` `name)` `{`
	`org.openstack4j.model.identity.Role` `role` `=` `os.identity().roles().create(name);`
`}`

Anyway when I run the program on the server I get this error:

`Exception in thread "main" ServerResponseException{message=Internal Server Error, status=500, status-code=INTERNAL_SERVER_ERROR}
	at org.openstack4j.core.transport.HttpExceptionHandler.mapException(HttpExceptionHandler.java:40)
	at org.openstack4j.core.transport.HttpExceptionHandler.mapException(HttpExceptionHandler.java:23)
	at org.openstack4j.core.transport.HttpEntityHandler.handle(HttpEntityHandler.java:50)
	at org.openstack4j.connectors.okhttp.HttpResponseImpl.getEntity(HttpResponseImpl.java:69)
	at org.openstack4j.openstack.internal.BaseOpenStackService$Invocation.execute(BaseOpenStackService.java:189)
	at org.openstack4j.openstack.internal.BaseOpenStackService$Invocation.execute(BaseOpenStackService.java:183)
	at org.openstack4j.openstack.identity.internal.RoleServiceImpl.create(RoleServiceImpl.java:121)
	at openstack.Identity.role_create(Identity.java:75)
	at openstack.openstackTest.main(openstackTest.java:21)`

When I create users I use Builders to create an User object and pass it to the create method. The create method for roles does not want a Role object but a String, that's why I just tried to pass the 'name' string. When the exception occurs I opened the RoleServiceImpl.java file to read the create method. That's the result: 

```
@Override
public Role create(String name) { 
    checkNotNull(name); 
    return post(KeystoneRole.class, uri("/OS-SADM/roles")).entity(KeystoneRole.builder().name(name).build()).execute(); 
}
```

So the Role is built inside this method. What's wrong? Can anyone help me to solve this problem and create a new Role?

Best regards,
Fabio

**Comment by gondor** [comment]
@FabioLiv 

Unfortunately OpenStack is hard to debug or know what the root cause is by submitting issues.  For example you may have an install without the extension for roles installed.  

What I suggest is running the OpenStack CLI python command to create the role and execute it in ```--debug``` mode.  I believe thats the flag, may have to look at help.   This will log the HTTP transactions and from that data I can vet the OpenStack4j implementation.

If the OpenStack CLI command fails then its your deployment of OpenStack thats missing this feature.  Hope that helps steer you in the right direction.

**Comment by FabioLiv** [comment]
@gondor 

Hi Jeremy,
thank you for the reply. I had already developed a web service with Python APIs and everybody works fine. I didn't exploit OpenStack CLI so I installed and used it in debug mode to create a role. The role was correctly created, this is the screenshot of the command output:
![schermata 2016-04-07 alle 16 13 18](https://cloud.githubusercontent.com/assets/18170093/14354205/c0c12010-fcdc-11e5-9e78-dc025b2a0ab1.png)

I attach also the text file:
[create_role_debug.txt](https://github.com/gondor/openstack4j/files/208543/create_role_debug.txt)

Best regards,
Fabio

**Comment by FabioLiv** [comment]
@gondor 

I used wireshark to capture HTTP packets in both cases (with OpenStack CLI and Openstack4j):
![schermata 2016-04-07 alle 17 36 26](https://cloud.githubusercontent.com/assets/18170093/14357103/adbdc36e-fce7-11e5-9dbc-697cc055a4a4.png)

Packets from 1475 to 1491 belong to OpenStack CLI command, packets 2903 and 2912 belong to Openstack4j command. I read the content POST packets, that's the result:
![schermata 2016-04-07 alle 17 35 42](https://cloud.githubusercontent.com/assets/18170093/14357162/eb151258-fce7-11e5-8421-03e418a9de22.png)
  
Packet on the right belongs to OpenStack4j. The object contains also the member "enabled", I didn't set it in my function.
`org.openstack4j.model.identity.Role role = os.identity().roles().create(name);`

So is it automatically set from the library? Could it be the problem if the version V2 of OpenStack only wants the parameter 'name' to create the role?

**Comment by gondor** [comment]
Thank you for all the feedback.  This should now be fixed.  I have checked in the fix and deployed a new 2.20-SNAPSHOT to Sonatype for your verification.

**Comment by FabioLiv** [comment]
@gondor 

It works!! 
![schermata 2016-04-08 alle 14 02 52](https://cloud.githubusercontent.com/assets/18170093/14386202/24e5e0ea-fda4-11e5-81f4-9ad945d7e25e.png)

Thank you very much! I'm waiting for the Maven configuration release ;)
